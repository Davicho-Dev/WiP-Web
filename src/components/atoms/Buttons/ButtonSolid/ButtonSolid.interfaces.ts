export interface IButtonSolidProps {
	label: string
	className?: string
	disabled?: boolean
	type?: 'button' | 'submit' | 'reset'
}
