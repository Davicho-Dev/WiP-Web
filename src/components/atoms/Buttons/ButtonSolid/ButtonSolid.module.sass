.button
  @apply bg-secondary text-primary w-4/5 h-10 rounded-3xl

  &:disabled
    @apply bg-neutral-300 text-neutral-500 cursor-not-allowed
