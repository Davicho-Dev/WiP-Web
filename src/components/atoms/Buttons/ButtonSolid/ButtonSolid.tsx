import { IButtonSolidProps } from './ButtonSolid.interfaces'

import styles from './ButtonSolid.module.sass'

export const ButtonSolid = ({
	className,
	label,
	type,
	disabled,
}: IButtonSolidProps): JSX.Element => {
	return (
		<button
			className={`${styles.button} ${className ?? ''}`}
			type={type}
			disabled={disabled}
		>
			{label}
		</button>
	)
}
