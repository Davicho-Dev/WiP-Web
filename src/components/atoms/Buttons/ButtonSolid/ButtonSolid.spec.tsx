import { render } from '@testing-library/react'

import { ButtonSolid } from './ButtonSolid'

describe('ButtonSolid test suite', () => {
	it('Should render correctly', () => {
		const label = 'Button Solid'

		const { getByText } = render(<ButtonSolid {...{ label }} />)

		expect(getByText(label)).toMatchSnapshot()
	})
})
