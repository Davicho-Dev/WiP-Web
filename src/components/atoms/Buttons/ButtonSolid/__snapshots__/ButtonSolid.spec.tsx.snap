// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSolid test suite Should render correctly 1`] = `
<button
  class="button "
>
  Button Solid
</button>
`;
